app/controllers/api/v1/agents_controller.rb

```diff
@@ -1,25 +1,27 @@
 class Api::V1::AgentsController < Api::BaseController
   before_action :fetch_agent, except: [:create, :index]
   before_action :check_authorization
-  before_action :build_agent, only: [:create]
+  before_action :find_user, only: [:create]
+  before_action :create_user, only: [:create]
+  before_action :save_account_user, only: [:create]
 
   def index
     @agents = agents
   end
 
   def destroy
-    @agent.destroy
+    @agent.account_user.destroy
     head :ok
   end
 
   def update
-    @agent.update!(agent_params)
-    render json: @agent
+    @agent.update!(agent_params.except(:role))
+    @agent.account_user.update!(role: agent_params[:role]) if agent_params[:role]
+    render 'api/v1/models/user.json', locals: { resource: @agent }
   end
 
   def create
-    @agent.save!
-    render json: @agent
+    render 'api/v1/models/user.json', locals: { resource: @user }
   end
```

Comment:
instead of using render in the controller wouldn't it be neater to use it in the view instead?

Reply:
We are using the `Jbuilder view` to render.